bin/tictactoe

```diff
@@ -1,3 +1,34 @@
 #!/usr/bin/env ruby
 
 require_relative '../config/environment'
+
+puts "Welcome to Tic Tac Toe"
+
+play = "Y"
+
+while play == "Y"
+
+  puts "Enter number of human players (0-2):"
+  humans = gets.chomp
+
+  if humans != "0"
+    puts "Player 1: 'X' or 'O'?"
+    token = gets.chomp
+    opponent_token = token == "X" ? "O" : "X"
+  end
+
+  case humans
+  when "0"
+    game = Game.new(Player::Computer.new("X"), Player::Computer.new("O"), Board.new)
+  when "1"
+    game = Game.new(Player::Human.new(token), Player::Computer.new(opponent_token), Board.new)
+  when "2"
+    game = Game.new(Player::Human.new(token), Player::Human.new(opponent_token), Board.new)
+  end
+
+  game.play
+
+  puts "Would you like to play again: 'Y' or 'N'?"
+  play = gets.chomp
+
+end
```

Comment:
Prefer to encapsulate CLI logic into a class. For example, the only code in your bin executable should be:

```ruby
#!/usr/bin/env ruby
       
require_relative '../config/environment'

TicTacToeCLI.new.call
```

And then you'd have a file: `lib/tic_tac_toe_cli.rb` with:

```ruby
class TicTacToeCLI
  def initialize
  end

  def call
   # Your game logic
  end
end
```

The code in your bin would generally just go in `call`.